src/VisualStudio/Core/Def/Implementation/TaskList/ProjectExternalErrorReporter.cs

```diff
@@ -224,10 +227,16 @@ private static DiagnosticSeverity GetDiagnosticSeverity(ExternalError error)
             string mappedFilePath, int mappedStartLine, int mappedStartColumn, int mappedEndLine, int mappedEndColumn,
             string originalFilePath, int originalStartLine, int originalStartColumn, int originalEndLine, int originalEndColumn)
         {
+            var document = id != null ? _workspace.CurrentSolution.GetDocument(id) : null;
+            var sourceSpan = document != null ?
+                document.GetSpanFromPositionAsync(originalStartLine, originalStartColumn, originalEndLine, originalEndColumn, CancellationToken.None).WaitAndGetResult(CancellationToken.None) :
```

Comment:
you shouldn't do this. by the way, are we supporting code action (including suppression) for build error? there is no way to know the build error is valid or not unless  you build before running LB like FxCop.

...

regardless of that, source span shouldn't be calculated here. getting source span is expensive since it require to bring in text.

it is done this way by design. you should call this
http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Diagnostics/DiagnosticData.cs,305508f892f5f654,references

if you want TextSpan from diagnostic from build, and you should do that only when it is needed.